File: GetItemViewType/app/src/main/java/com/nipun/getitemviewtype/MainActivity.java
package com.nipun.getitemviewtype;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.nipun.getitemviewtype.adapter.ChatAdapter;
import com.nipun.getitemviewtype.model.BaseModel;
import com.nipun.getitemviewtype.model.receiver;
import com.nipun.getitemviewtype.model.sender;

import java.sql.Timestamp;
import java.text.Format;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private List<BaseModel> baseModels = new ArrayList<>();
    private RecyclerView recyclerView;
    private EditText etChat;
    private EditText etSelect;
    private Button btSend;
    private ChatAdapter chatAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
        setRecyclerView();
        chat();

    }

    private void chat() {
        if(etSelect.getText().toString().equals("1")){
            baseModels.add(new sender(etChat.getText().toString()));

        }else if(etSelect.getText().toString().equals("2")){
            baseModels.add(new receiver(etChat.getText().toString(),getFormateDate()));
        }
        chatAdapter.notifyDataSetChanged();

    }

    private void setRecyclerView() {

        LinearLayoutManager layoutManager= new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        chatAdapter= new ChatAdapter(baseModels);
        recyclerView.setAdapter(chatAdapter);
    }

    private void initView() {

        recyclerView = findViewById(R.id.rcyclerView);
        etChat = findViewById(R.id.etChatMessage);
        etSelect = findViewById(R.id.etUserSelect);
        btSend = findViewById(R.id.btnSend);
        btSend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                chat();
            }
        });
    }

    public String getFormateDate(){
        Timestamp timestamp = new Timestamp((System.currentTimeMillis()));
        Date date = new Date(timestamp.getTime());
        Format format = new SimpleDateFormat("HH.mm");
        return format.format(date);
    }
}

File: Evaluation5/app/src/main/java/com/nipun/evaluation5/MainActivity.kt
package com.nipun.evaluation5

import android.media.AudioManager
import android.media.MediaPlayer
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import retrofit2.Call
import retrofit2.Callback
import java.io.IOException

class MainActivity : AppCompatActivity() , OnClickListner {

    var button: Button? = null
    var recyclerView: RecyclerView? = null
    var responseList: Response? = null
    var dataAdapter: DataAdapter? = null
    var editText: EditText? = null
    val mediaPlayer = MediaPlayer()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        button = findViewById<Button>(R.id.btn)
        recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        editText = findViewById<EditText>(R.id.editSearch)
        button!!.setOnClickListener { callApi() }
    }

    open fun callApi() {
        val str = editText!!.text.toString()
        val apiService = Network.getInstance().create(
            ApiService::class.java
        )
        apiService.getUser(str).enqueue(object : Callback<Response?> {
            override fun onResponse(
                call: Call<Response?>,
                response: retrofit2.Response<Response?>
            ) {
                setRecyclerView(responseList!!)
                if (response.body() != null) {
                    responseList = response.body()
                    dataAdapter!!.updateList(responseList)
                }
            }

            override fun onFailure(call: Call<Response?>, t: Throwable) {}
        })
    }

    open fun setRecyclerView(responseList: Response) {
        dataAdapter = DataAdapter(responseList, this)
        dataAdapter!!.notifyDataSetChanged()
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = linearLayoutManager
        recyclerView!!.adapter = dataAdapter
    }

    override fun playMusic(pos: Int, result: Result) {
        mediaPlayer.reset()
        if (mediaPlayer.isPlaying) {
            mediaPlayer.stop()
        }
        mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC)
        try {
            mediaPlayer.setDataSource(result.previewUrl)
            mediaPlayer.prepare()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        mediaPlayer.start()
    }

    override fun pauseMusic() {
        mediaPlayer.stop()
    }

    override fun delete(pos: Int) {
        /*responseList!!.results.
        dataAdapter!!.updateList(responseList)*/
    }

}